SixTrack/sixtrack.s

```diff
@@ -57998,7 +58009,7 @@ c$$$            endif
 !     for fma_norm_flag = 0 use physical coordinates x,x',y,y',sig,dp/p
 !         fma_norm_flag > 0 use normalized coordinates
             do l=1,napx ! loop over particles
-              do m=1,3 ! loop over modes (hor.,vert.,long.)
+              do m=1,num_modes ! loop over modes (hor.,vert.,long.)
```

Comment:
This doesn't work as you then have to use the 4D matrix instead of the 6D matrix, otherwise the normalization is not correct. I would suggest to implement this properly and use for 4D tracking the 4D matrix and 6D tracking the 6D matrix.